src/In.ProjectEKA.HipService/DataFlow/DataFlowClient.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Hl7.Fhir.Serialization;
+using In.ProjectEKA.HipLibrary.Patient;
+using In.ProjectEKA.HipService.Logger;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace In.ProjectEKA.HipService.DataFlow
+{
+    public class DataFlowClient
+    {
+        private readonly ICollect collect;
+        private readonly HttpClient httpClient;
+
+        public DataFlowClient(ICollect collect, HttpClient httpClient)
+        {
+            this.collect = collect;
+            this.httpClient = httpClient;
+        }
+
+        public async Task HandleMessagingQueueResult(HipLibrary.Patient.Model.DataRequest dataRequest)
+        {
+            (await collect.CollectData(dataRequest))
+                .Map(async entries =>
+                {
+                    var serializer = new FhirJsonSerializer(new SerializerSettings());
+                    var healthRecordEntries = entries.Bundles
+                        .Select(bundle => new Entry(
+                            serializer.SerializeToString(bundle),
+                            "application/json",
+                            "MD5"))
+                        .ToList();
+                    await SendDataToHiu(new DataResponse(dataRequest.TransactionId, healthRecordEntries),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5014227)